examples/chip-tool/commands/common/CHIPCommand.cpp

```diff
@@ -172,28 +184,33 @@ CHIP_ERROR CHIPCommand::InitializeCommissioner(std::string key, chip::FabricId f
     VerifyOrReturnError(icac.Alloc(chip::Controller::kMaxCHIPDERCertLength), CHIP_ERROR_NO_MEMORY);
     VerifyOrReturnError(rcac.Alloc(chip::Controller::kMaxCHIPDERCertLength), CHIP_ERROR_NO_MEMORY);
 
-    chip::MutableByteSpan nocSpan(noc.Get(), chip::Controller::kMaxCHIPDERCertLength);
-    chip::MutableByteSpan icacSpan(icac.Get(), chip::Controller::kMaxCHIPDERCertLength);
-    chip::MutableByteSpan rcacSpan(rcac.Get(), chip::Controller::kMaxCHIPDERCertLength);
-
     chip::Crypto::P256Keypair ephemeralKey;
-    ReturnLogErrorOnFailure(ephemeralKey.Initialize());
 
-    // TODO - OpCreds should only be generated for pairing command
-    //        store the credentials in persistent storage, and
-    //        generate when not available in the storage.
-    ReturnLogErrorOnFailure(mCommissionerStorage.Init(key.c_str()));
-    ReturnLogErrorOnFailure(mCredIssuerCmds->InitializeCredentialsIssuer(mCommissionerStorage));
-    ReturnLogErrorOnFailure(mCredIssuerCmds->GenerateControllerNOCChain(mCommissionerStorage.GetLocalNodeId(), fabricId,
-                                                                        ephemeralKey, rcacSpan, icacSpan, nocSpan));
+    if (fabricIndex != chip::kUndefinedFabricIndex)
+    {
+
+        // TODO - OpCreds should only be generated for pairing command
+        //        store the credentials in persistent storage, and
+        //        generate when not available in the storage.
+        ReturnLogErrorOnFailure(mCommissionerStorage.Init(key.c_str()));
+        ReturnLogErrorOnFailure(mCredIssuerCmds->InitializeCredentialsIssuer(mCommissionerStorage));
+
+        chip::MutableByteSpan nocSpan(noc.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan icacSpan(icac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan rcacSpan(rcac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+
+        ReturnLogErrorOnFailure(ephemeralKey.Initialize());
+        ReturnLogErrorOnFailure(mCredIssuerCmds->GenerateControllerNOCChain(mCommissionerStorage.GetLocalNodeId(), fabricIndex,
+                                                                            ephemeralKey, rcacSpan, icacSpan, nocSpan));
+        commissionerParams.ephemeralKeypair = &ephemeralKey;
+        commissionerParams.controllerRCAC   = rcacSpan;
+        commissionerParams.controllerICAC   = icacSpan;
+        commissionerParams.controllerNOC    = nocSpan;
+    }
 
     commissionerParams.storageDelegate                = &mCommissionerStorage;
-    commissionerParams.fabricIndex                    = static_cast<chip::FabricIndex>(fabricId);
+    commissionerParams.fabricIndex                    = fabricIndex;
```

Comment:
This is not know a-priori. Overall, fabricIndex should never be part of commissionerParams, and rather be self-derived from the fabric table position of a given root+fabric ID. This argument should disappear, and the default should be `kUndefinedFabricIndex` until we know

Reply:
Done.

chip-tool now just passes the operational keys and cert chain, and the commissioner / controller init 'figures it out' by comparing to what's in the fabric table.

If a matching fabric is found, it is updated.  If none is found, a new one is created.

This functions the same as what we have now, but no longer requires fabric index and fabric ID to be specified to the commissioner for init, so is much less brittle.